lib/ae_mdw/names.ex

```diff
@@ -770,13 +761,14 @@ defmodule AeMdw.Names do
     {serialize_nested_cursor(prev_cursor), bi_txi_idxs, serialize_nested_cursor(next_cursor)}
   end
 
-  defp render_previous(state, name, last_gen, last_micro_time, opts) do
-    name
-    |> Stream.unfold(fn
-      Model.name(previous: nil) -> nil
-      Model.name(previous: previous) -> {previous, previous}
+  defp render_previous(state, plain_name, last_gen, last_micro_time, opts) do
+    key_boundary = {{plain_name, @min_int}, {plain_name, @max_int}}
+
+    state
+    |> Collection.stream(Model.PreviousName, :backward, key_boundary, nil)
+    |> Enum.map(fn Model.previous_name(name: name) ->
```

Comment:
This needs a `fetch!` to have the `name` after the `index`.

Reply:
Fixed, thanks